packages/connect/src/components.ts

```diff
@@ -0,0 +1,229 @@
+import errCode from 'err-code'
```

Comment:
Use `Components` pattern from `libp2p` also for `connect` modules

Reply:
Some documentation would be very useful here